localstack/services/apigateway/provider.py

```diff
@@ -844,6 +972,17 @@ def filter_function(item):
 # ---------------
 
 
+def get_moto_rest_api(context: RequestContext, rest_api_id: str) -> MotoRestAPI:
```

Comment:
we do have this method `self._get_moto_backend(context)` on the provider similar to this one. This one seems to handle mis-lookups, should we highlight that and remove usages of the former?

Reply:
You're right, the `self._get_moto_backend(context)` was from my first PR, but not sure it still makes sense now, this new method handles a bit more. If we ever need to access the moto backend to access other kind of resources and not RestAPI, we can create a new method for that resource with the proper exception message. Thanks for picking this up!